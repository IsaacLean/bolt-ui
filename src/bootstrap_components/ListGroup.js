/*
 * BOOTSTRAP LIST GROUP COMPONENT
 * https://v4-alpha.getbootstrap.com/components/list-group
 */

import React from 'react';

export default class ListGroup extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    let listGroupItems = [];

    if(this.props.listGroupItems && this.props.listGroupItems.constructor === Array) {
      listGroupItems = this.props.listGroupItems.map((listGroupItem, i) => <li className="list-group-item" key={i}>
        {listGroupItem}
      </li>);
    }

    return <ul className="list-group">
      {listGroupItems}
    </ul>;
  }
}
