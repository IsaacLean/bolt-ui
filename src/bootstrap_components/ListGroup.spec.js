import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import ListGroup from './ListGroup';

test('Empty ListGroup snapshot', () => {
  const component = shallow(<ListGroup />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

test('ListGroup with listGroupItems.length == 1 snapshot', () => {
  const component = shallow(<ListGroup listGroupItems={['Item 1']} />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});


test('ListGroup with listGroupItems.length > 1 snapshot', () => {
  const component = shallow(<ListGroup listGroupItems={['Item 1', 'Item 2']} />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});
