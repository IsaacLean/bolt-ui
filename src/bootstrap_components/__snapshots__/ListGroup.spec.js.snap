// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty ListGroup snapshot 1`] = `
<ul
  className="list-group"
/>
`;

exports[`ListGroup with listGroupItems.length == 1 snapshot 1`] = `
<ul
  className="list-group"
>
  <li
    className="list-group-item"
  >
    Item 1
  </li>
</ul>
`;

exports[`ListGroup with listGroupItems.length > 1 snapshot 1`] = `
<ul
  className="list-group"
>
  <li
    className="list-group-item"
  >
    Item 1
  </li>
  <li
    className="list-group-item"
  >
    Item 2
  </li>
</ul>
`;
